build.rs

```diff
@@ -0,0 +1,52 @@
+fn main() {
+    #[cfg(target_os = "windows")]
+    download_winpcap_sdk()
+}
+
+#[cfg(target_os = "windows")]
+fn download_winpcap_sdk() {
+    use http_req::request;
+    use std::env;
+    use std::fs::File;
+    use std::io::prelude::*;
+
+    let out_dir = env::var("OUT_DIR").unwrap();
+
+    let mut reader = Vec::new();
+    let res = request::get(
+        "https://nmap.org/npcap/dist/npcap-sdk-1.05.zip",
+        &mut reader,
+    )
+    .unwrap();
+
+    println!("Status: {} {}", res.status_code(), res.reason());
+    println!("Reader len: {}", reader.len());
+
+    let mut pcapzip = File::create(format!("{}{}", out_dir, "/npcap.zip")).unwrap();
+    pcapzip.write_all(reader.as_slice()).unwrap();
+    pcapzip.flush().unwrap();
+
+    pcapzip = File::open(format!("{}{}", out_dir, "/npcap.zip")).unwrap();
+
+    #[cfg(target_arch = "x86_64")]
+    let lib_path = "Lib/x64/Packet.lib";
+    #[cfg(target_arch = "x86")]
+    let lib_path = "Lib/Packet.lib";
+
+    let mut zip_archive = zip::ZipArchive::new(pcapzip).unwrap();
+    let mut pcaplib = match zip_archive.by_name(lib_path) {
+        Ok(pcaplib) => pcaplib,
+        Err(err) => {
+            panic!(err);
+        }
+    };
+
+    let mut pcaplib_bytes = Vec::new();
+    pcaplib.read_to_end(&mut pcaplib_bytes).unwrap();
+
+    let mut pcaplib_file = File::create(format!("{}{}", out_dir, "/Packet.lib")).unwrap();
+    pcaplib_file.write_all(pcaplib_bytes.as_slice()).unwrap();
+    pcaplib_file.flush().unwrap();
+
+    println!("cargo:rustc-link-search=native={}", out_dir);
```

Comment:
So, going over: https://doc.rust-lang.org/cargo/reference/build-scripts.html#cargorustc-link-searchkindpath
I understand this kind of thing is discouraged. Is there a way to install this on the system level, or is this something we'd want the user to do manually/have the package manager do it?

Reply:
This is a static library, therefore I think that the rust docs warning does not apply here. Unfortunately there is no such thing as package manager on windows (at least not when you think about apt-get install lib<something>-dev for C libs and headers). 
Another [rust library](https://github.com/ebfull/pcap) for interacting with libpcap does the same but in its  [Github Action](https://github.com/ebfull/pcap/blob/master/.github/workflows/rust.yml). I think that build.rs approach is better, as it allows any developer to build windows version locally.
The Windows user will need to install NPcap driver manually for bandwidth to work properly, but that is a standard prerequisite for anything which sniffs the network on Windows (like wireshark).